include/geopackage/SQLite.hpp

```diff
@@ -0,0 +1,273 @@
+#ifndef NGEN_GEOPACKAGE_SQLITE_H
+#define NGEN_GEOPACKAGE_SQLITE_H
```

Comment:
I'm mildly concerned about naming a file we own simply `SQLite.hpp`, since that could look like it's code from the library in stack traces or whatever. Would you mind prefixing its name?

Reply:
In that vein, I notice that all of the class and function names declared here are prefixed with `sqlite_`. It would be preferable to have everything in a namespace like `ngen::sqlite`, and then the individual prefixes could be dropped.

Reply:
File name change is implemented in 31fb4b1. I am gong to hold off on encasing the sqlite definitions inside an `ngen` namespace, since I think there is a larger reorg issue there with the other sub-libraries in NGen, and that should be addressed separately (I think)